kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -159,12 +159,30 @@ function redirectToExploreChannel(store) {
       if (currentChannelId) {
         store.dispatch('SET_CURRENT_CHANNEL', currentChannelId);
         cookiejs.set('currentChannel', currentChannelId);
-        router.go({
-          name: constants.PageNames.EXPLORE_CHANNEL,
-          params: {
-            channel_id: currentChannelId,
-          },
-        });
+        if (cookiejs.get('currentExploreType') === 'topic') {
+          router.go({
+            name: constants.PageNames.EXPLORE_TOPIC,
+            params: {
+              channel_id: currentChannelId,
+              id: cookiejs.get('currentExploreId'),
+            },
+          });
+        } else if (cookiejs.get('currentExploreType') === 'content') {
+          router.go({
+            name: constants.PageNames.EXPLORE_CONTENT,
+            params: {
+              channel_id: currentChannelId,
+              id: cookiejs.get('currentExploreId'),
+            },
+          });
+        } else {
+          router.go({
+            name: constants.PageNames.EXPLORE_CHANNEL,
+            params: {
+              channel_id: currentChannelId,
+            },
+          });
+        }
```

Comment:
@indirectlylit Note that this is not inside the _getCurrentChannelId function, it is chained from the returned Promise from that function.